test/hisi_zip_test/test_sva_perf.c

```diff
@@ -1020,57 +1056,59 @@ static int run_self_test(void)
 		.total_len		= 8192 * 10,
 		.compact_run_num	= 1000,
 	};
-	int ret, f_ret = 0;
+	int i, ret, f_ret = 0;
 
 	printf("Start to run self test!\n");
-	/* SYNC mode */
 	ret = test_sw_dfl_sw_ifl();
 	if (ret)
 		printf("Fail on running test_sw_dfl_sw_ifl():%d\n", ret);
 	f_ret |= ret;
-	ret = test_sw_dfl_hw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_sw_dfl_hw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_sw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_sw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_hw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_hw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_perf(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_perf():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_ifl_perf(&opts);
-	if (ret)
-		printf("Fail on running test_hw_ifl_perf():%d\n", ret);
-	f_ret |= ret;
-
-	/* ASYNC mode */
-	opts.sync_mode = 1;
-	ret = test_sw_dfl_hw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_sw_dfl_hw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_sw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_sw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_hw_ifl(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_hw_ifl():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_dfl_perf(&opts);
-	if (ret)
-		printf("Fail on running test_hw_dfl_perf():%d\n", ret);
-	f_ret |= ret;
-	ret = test_hw_ifl_perf(&opts);
-	if (ret)
-		printf("Fail on running test_hw_ifl_perf():%d\n", ret);
-	f_ret |= ret;
+	for (i = 0; i < 5; i++) {
+		switch (i) {
+		case 0:
+			opts.sync_mode = 0;
+			opts.block_size = 8192; opts.total_len = 8192 * 10;
+			break;
+		case 1:
+			opts.sync_mode = 1; 	opts.poll_num = 1;
+			opts.block_size = 8192; opts.total_len = 8192 * 10;
+			break;
+		case 2:
+			opts.sync_mode = 1; 	opts.poll_num = 2;
+			opts.block_size = 8192; opts.total_len = 8192 * 10;
+			break;
+		case 3:
+			opts.sync_mode = 1;	opts.poll_num = 4;
+			opts.block_size = 8192;	opts.total_len = 8192 * 10;
+			break;
+		case 4:
+			opts.sync_mode = 1; 	opts.poll_num = 8;
+			opts.block_size = 8192; opts.total_len = 8192 * 10;
+			break;
```

Comment:
better to make them dynamically, and use a script to change different parameters :)